src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.c

```diff
@@ -46,80 +77,1172 @@ const ActorInit En_Kakasi_InitVars = {
+ * before actually teaching
+ */
+void EnKakasi_OcarinaRemark(EnKakasi* this, GlobalContext* globalCtx) {
+    if (func_80152498(&globalCtx->msgCtx) == 5 && func_80147624(globalCtx) != 0) {
+        func_80152434(globalCtx, 0x35);
+        this->unkState1A8 = 0;
+        if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+            ActorCutscene_Stop(0x7C);
+            ActorCutscene_SetIntentToPlay(this->actorCutscenes[0]);
+            this->actionFunc = EnKakasi_TeachingSong;
+
+        } else if (ActorCutscene_GetCanPlayNext(this->actorCutscenes[0]) == 0) {
+            ActorCutscene_SetIntentToPlay(this->actorCutscenes[0]);
+            this->actionFunc = EnKakasi_TeachingSong;
+
+        } else {
+            this->unkState1A8 = 1;
+            ActorCutscene_StartAndSetUnkLinkFields(this->actorCutscenes[0], &this->actor);
+            this->cutsceneCamId = ActorCutscene_GetCurrentCamera(this->actor.cutscene);
+            Math_Vec3f_Copy(&this->unk22C, &this->actor.home.pos);
+            this->actionFunc = EnKakasi_TeachingSong;
+        }
+    }
+}
+
+void EnKakasi_TeachingSong(EnKakasi* this, GlobalContext* globalCtx) {
+    Vec3f tempVec;
+
+    EnKakasi_CheckPlayerPosition(this, globalCtx);
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.home.rot.y, 1, 3000, 0);
+    if (this->unkState1A8 == 0) {
+        if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+            ActorCutscene_Stop(0x7C);
+            ActorCutscene_SetIntentToPlay(this->actorCutscenes[0]);
+            return;
+        }
+        if (ActorCutscene_GetCanPlayNext(this->actorCutscenes[0]) == 0) {
+            ActorCutscene_SetIntentToPlay(this->actorCutscenes[0]);
+            return;
+        }
+        ActorCutscene_StartAndSetUnkLinkFields(this->actorCutscenes[0], &this->actor);
+        this->cutsceneCamId = ActorCutscene_GetCurrentCamera(this->actor.cutscene);
+        Math_Vec3f_Copy(&this->unk22C, &this->actor.home.pos);
+        this->unkState1A8 = 1;
+        this->unkState1A8 = 1;
+    }
+
+    if (this->unkState1A8 == 1) {
+        this->unk22C.y = this->actor.home.pos.y + 50.0f;
+        this->unk238.x = D_80971DCC[this->unk190].x;
+        this->unk238.y = D_80971DCC[this->unk190].y;
+        this->unk238.z = D_80971DCC[this->unk190].z;
+
+        Math_Vec3f_Copy(&tempVec, &this->unk238);
+        OLib_DbCameraVec3fSum(&this->actor.home, &tempVec, &this->unk238, 1);
+        Math_Vec3f_Copy(&this->unk244, &this->unk22C);
+        Math_Vec3f_Copy(&this->unk214, &this->unk238);
+        Math_Vec3f_Copy(&this->unk220, &this->unk244);
+        func_8096FAAC(this, globalCtx);
+        func_8096FBB8(this, globalCtx);
+
+        // song failed
+        if (globalCtx->msgCtx.unk1202A == 4) {
+            this->unk190 = 0;
+            this->unkCounter1A4 = 0;
+            ActorCutscene_Stop(this->actorCutscenes[0]);
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_YASE_DEAD);
+            if (!this) {}
```

Comment:
```suggestion
            if (this) {}
```
`if (1) {}` would be even better